php/blocks/class-utils.php

```diff
@@ -0,0 +1,141 @@
+<?php
+/**
+ * Helper functions for the Block_Lab plugin.
+ *
+ * These are publicly accessible via a magic method, like block_lab->get_template_locations().
+ * So these methods should generally be 'getter' functions, and should not affect the global state.
+ *
+ * @package Block_Lab
+ */
+
+namespace Block_Lab\Blocks;
+
+use Block_Lab\Component_Abstract;
+
+/**
+ * Class Utils
+ */
+class Utils extends Component_Abstract {
+
+	/**
+	 * Not implemented, as this class only has utility methods.
+	 */
+	public function register_hooks() {}
+
+	/**
+	 * Gets whether a valid Pro license has been activated on this site.
+	 *
+	 * @return bool
+	 */
+	public function is_pro() {
+		return $this->plugin->admin->license->is_valid();
+	}
+
+	/**
+	 * Gets an array of possible template locations.
+	 *
+	 * @param string $name The name of the block (slug as defined in UI).
+	 * @param string $type The type of template to load. Typically block or preview.
+	 *
+	 * @return array
+	 */
+	public function get_template_locations( $name, $type = 'block' ) {
+		return array(
+			"blocks/{$name}/{$type}.php",
+			"blocks/{$type}-{$name}.php",
+			"blocks/{$type}.php",
+		);
+	}
+
+	/**
+	 * Gets an array of possible stylesheet locations.
+	 *
+	 * @param string $name The name of the block (slug as defined in UI).
+	 * @param string $type The type of template to load. Typically block or preview.
+	 *
+	 * @return array
+	 */
+	public function get_stylesheet_locations( $name, $type = 'block' ) {
+		return array(
+			"blocks/{$name}/{$type}.css",
+			"blocks/css/{$type}-{$name}.css",
+			"blocks/{$type}-{$name}.css",
+		);
+	}
+
+	/**
+	 * Locates templates.
+	 *
+	 * Works similar to `locate_template`, but allows specifying a path outside of themes
+	 * and allows to be called when STYLESHEET_PATH has not been set yet. Handy for async.
+	 *
+	 * @param string|array $template_names Templates to locate.
+	 * @param string       $path           (Optional) Path to locate the templates first.
+	 * @param bool         $single         `true` - Returns only the first found item. Like standard `locate_template`
+	 *                                     `false` - Returns all found templates.
+	 *
+	 * @return string|array
+	 */
+	public function locate_template( $template_names, $path = '', $single = true ) {
```

Comment:
This is simply moved from `helpers.php`, not changed at all. It'd be good to have unit tests for this soon.

Reply:
It now has a unit test.

Reply:
Update: there's now a unit test for it.